.github/workflows/promote.yaml

```
@@ -1,25 +1,35 @@
-name: Promote charm to other tracks and channels
+name: Promote charm to default track, standard risk levels.
 
 on:
   workflow_dispatch:
     inputs:
-      destination-channel:
-        description: 'Destination Channel, e.g. latest/candidate'
-        required: true
-      origin-channel:
-        description: 'Origin Channel, e.g. latest/edge'
+      channel-promotion:
+        description: 'Channel Promotion, e.g. latest/edge -> latest/candidate'
         required: true
+        type: choice
+        options:
+          - 'latest/edge -> latest/candidate'
+          - 'latest/candidate -> latest/stable'
 
 jobs:
   promote-charm:
     name: Promote charm
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
+      - name: Set channels
+        id: set-channels
+        run: |
+          channel_promotion="${{ github.event.inputs.channel-promotion }}"
+          origin=$(echo "$channel_promotion" | sed 's/\s*->.*//')
+          destination=$(echo "$channel_promotion" | sed 's/.*->\s*//')
```

suggestion: I think awk would be more readable here
```suggestion
          fs=' -> '
          origin=$(echo "$channel_promotion" | awk -F "$fs" '{print $1}')
          destination=$(echo "$channel_promotion" | awk -F "$fs" '{print $2}')
```

---

Probably, but is it ok if we refactor later?  This simply copies our existing [template from the automation repo](https://github.com/canonical/solutions-engineering-automation/blob/2ee73f75536bb936a251c37817db4c1ba0892621/terraform-plans/templates/github/charm_promote.yaml.tftpl) - ideally we would do any refactorings from the automation repo so we can keep them all consistent across projects.